src/types/class.ts

```diff
@@ -0,0 +1,2 @@
+type StringOrBooleanObject = string | Record<string, boolean>;
+export type ClassType = StringOrBooleanObject | StringOrBooleanObject[];
```

Comment:
I'd name it like `ComponentCssClass` instead of `ClassType`, mostly to make it less generic and less similar to javascript's "real" Classes